### PR TITLE
Update go.mod

### PR DESCRIPTION
**Pull request by freshe4qa**
important patches

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Review by evan-forbes — CHANGES_REQUESTED** [review]
thanks for attempting this @freshe4qa, but I would like some clarity on why some of these changes were made.

any changes to the go.mod should be done via `go mod tidy` and not manually changes unless we have a specific reason to change.

I see changes relating to grpc and the grpc-gateway that puts us ahead of our fork of the cosmos-sdk, which we have avoided doing in the past, why should we do them now?

File: go.mod [hunk 1]
```diff
@@ -10,30 +10,30 @@ require (
...
-	golang.org/x/sys v0.2.0 // indirect
-	golang.org/x/term v0.2.0 // indirect
-	google.golang.org/genproto v0.0.0-20221024183307-1bc688fe9f3e
-	google.golang.org/grpc v1.50.1
+	golang.org/x/crypto v0.5.0 // indirect
+	golang.org/x/net v0.5.0 // indirect
+	golang.org/x/sys v0.4.0 // indirect
+	golang.org/x/term v0.4.0 // indirect
+	google.golang.org/genproto v0.0.0-20230113154510-dbe35b8444a5
+	google.golang.org/grpc v1.52.0
```
[the remark this answers is not in the document]
Comment (freshe4qa) [reply go.mod:24]: I know. I didn't have time to explain. Did an update for you to update what you need. And what you don't need to update, return to the previous version.

**Comment by freshe4qa** [comment]
It would be good to review go.sum

**Comment by rootulp** [comment]
hey @freshe4qa thanks for attempting to update our dependencies. As @evan-forbes said, celestia-app delays the upgrade of certain dependencies until cosmos-sdk performs them. The usual flow is:
1. cosmos/cosmos-sdk releases a new version
2. celestiaorg/cosmos-sdk pulls in upstream changes
3. celestia-app bumps its dependency of celestiaorg/cosmos-sdk and simultaneously takes updates for things like grpc

So let's close this PR. If there's a dependency you think needs to be upgraded ASAP (i.e. a security vulnerability) then please reach out to us at security@celestia.org

**Pull request closed without merging**